streams/src/main/java/org/apache/kafka/streams/StreamsMetrics.java

```diff
@@ -27,4 +28,17 @@
     Sensor addLatencySensor(String scopeName, String entityName, String operationName, String... tags);
```

Comment:
You are correct, previously this made sense since we did not expose adding arbitrary sensors onto the base metrics registry, but now that we do, having this in the interface stands out as odd

That said there seems to be a lot of boiler plate code around tags, parsing and logging. 



 // first add the global operation metrics if not yet, with the global tags only
            Sensor parent = metrics.sensor(scopeName + "-" + operationName);
            addLatencyMetrics(metricGroupName, parent, "all", operationName, this.metricTags);

            // add the store operation metrics with additional tags
            Sensor sensor = metrics.sensor(scopeName + "-" + entityName + "-" + operationName, parent);
            addLatencyMetrics(metricGroupName, sensor, entityName, operationName, tagMap);

What do these tags buy us ?  (it is not clear :) )

If we can get rid of the tags, then does it make sense to have an a base implementation of the StreamsMetrics interface, because with the exception of the actual sensors contained in them, the two implementations start to look very similar 

 protected class ProcessorNodeStreamsMetrics implements StreamsMetrics ....

  protected class ThreadStreamsMetrics implements StreamsMetrics ....